### ISSUE TITLE
EC2 instance is unable to login to vault using pkcs7 

### ISSUE THREAD

**Issue by secretlyvogon** [hashicorp/vault#12591]
**Describe the bug**
I have set up my vault server with EC2 login as suggested in the docs [here](https://registry.terraform.io/modules/hashicorp/vault/aws/latest/examples/vault-ec2-auth). I use the curl request to login to Vault but I get the following response:
```
{"errors":[]}
```
I tried the login with vault CLI from the instance where the server was hosted using the role and pkcs7 and it was successful. HTTP API is not working.

**Expected behavior**
According to the docs, I should be getting a JSON string with the client token and nonce. 

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.6.3
* Vault CLI Version (retrieve with `vault version`): Vault v1.6.3 (b540be4b7ec48d0dd7512c8d8df9399d6bf84d76)
* Server Operating System/Architecture: Ubuntu 18.04.5 LTS (GNU/Linux 5.4.0-1046-gcp x86_64)

Vault server configuration file(s):

```hcl
storage "raft" {
  path    = "./data"
  node_id = "node1"
}

listener "tcp" {
  address     = "127.0.0.1:8200"
  tls_disable = "true"
}

api_addr = "http://127.0.0.1:8200"
cluster_addr = "https://127.0.0.1:8201"
ui = true
```

**Comment by hsimon-hashicorp** [comment]
Hi @secretlyvogon! Can you provide us with a copy of your curl command, perhaps with a -v switch for extra verbosity? It'd be helpful to see what's being attempted and what's being provided back. Thanks!

**Comment by secretlyvogon** [comment]
Thanks for responding @hsimon-hashicorp . I followed the [HTTP API documentation](https://www.vaultproject.io/api/auth/aws#login) and used the following curl command:

```
curl -v \
    --request POST \
    --data @payload.json \
    https://vault.example.com/v1/auth/aws/login
```
My `payload.json` consisted of:
```
{
    'role' : 'test-role'
    'pkcs7' : <pkcs7 value obtained from ec2>
}
```
On performing the curl command with the verbosity option, I get the following output (removed TLS handshake messages)
```
> POST /v1/aws/auth/login HTTP/1.1
> Host: vault.example.com
> User-Agent: curl/7.71.1
> Accept: */*
> Content-Length: 1195
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 1195 out of 1195 bytes
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Cache-Control: no-store
< Content-Type: application/json
< Date: Fri, 24 Sep 2021 13:34:17 GMT
< Content-Length: 36
< 
{"errors":["missing client token"]}
```
I mentioned in the original post that I was getting an empty error string which was true. I don't know how I am getting the `missing client token` message now. Even so, I don't think it is a valid response since I should be getting the token in the success response.

**Comment by hsimon-hashicorp** [comment]
@secretlyvogon I'm assuming your pkcs7 entry in your payload.json has all of the newlines, etc, removed? I'm looking at the **Login** section of the [API document for the AWS auth](https://www.vaultproject.io/api/auth/aws) method and you're correct, that should be all you need if the certificate is set up correctly.

**Comment by secretlyvogon** [comment]
Yes, all new line characters in the pkcs7 entry have been removed.

**Comment by secretlyvogon** [comment]
@hsimon-hashicorp I upgraded vault binary to version 1.8.3 and I see the same issue. Any ideas?

**Comment by hsimon-hashicorp** [comment]
Hi @secretlyvogon - was Hamid's response up there helpful? Were you able to gather more information for us? Thanks in advance! :)

**Comment by secretlyvogon** [comment]
Hi, sorry for the late reply. I think the `{"errors":["missing client token"]}` message was because of the TOFU mechanism @hghaf099 described. However, I noticed that I was still getting `{"errors":[]}` in my setup on the first login. 

My vault server was hosted in GCP and the instance I was trying to log in from AWS EC2 instance. I later setup the vault server in another EC2 instance and didn't see this issue again. Could GCP hosting for EC2 login be the reason for the empty error string response?
